src/main/fc/fc_msp.c

```diff
@@ -2038,31 +2042,53 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 #if defined(OSD) || defined (USE_OSD_SLAVE)
     case MSP_SET_OSD_CONFIG:
         {
-            const uint8_t addr = sbufReadU8(src);
-            // set all the other settings
-            if ((int8_t)addr == -1) {
+            const uint8_t addr1 = sbufReadU8(src);
+            const uint8_t addr2 = sbufReadU8(src);
```

Comment:
Having a closer look at this, and considering that it's unlikely to happen that there are more than 127 OSD items or OSD stats, my recommendation would be to keep the protocol as it is, and slot in the OSD stats at addresses with an offset of 128. This way, the new firmware will keep working with clients that do implement the existing protocol. Also, n.b. that you are changing the indicator for 'OSD settings' from -1 to 0, thus breaking the protocol badly if a user was ever to use an old configurator with a new firmware.